Mgdapi 1192 3

laurafitzgerald opened the pull request:
See https://github.com/integr8ly/integreatly-operator/pull/1604 for details but use the branch on integreatly as the feature branch

laurafitzgerald commented:
/hold

laurafitzgerald commented:
@psturc would you mind creating a pr to fix the test case lint? or should i push a new commit?

laurafitzgerald commented:
/test e2e
/test e2e-rhoam

laurafitzgerald commented:
/test rhoam-e2e

laurafitzgerald commented:
/test e2e
/test e2e-rhoam

laurafitzgerald commented:
/test rhoam-e2e

laurafitzgerald commented:
/test images

KevFan commented on pkg/products/marin3r/rateLimitService.go:
```diff
@@ -354,7 +368,16 @@ func (r *RateLimitServiceReconciler) getRedisSecret(ctx context.Context, client
+	ratelimitconfig := yamlRoot{}
+	err := yaml.Unmarshal([]byte(c.Data[RateLimitingConfigMapDataName]), &ratelimitconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("error unmarshalling ratelimiting config from configmap '%s'", c.Name))
```
Very small but could we use `fmt.Errorf` instead ? I think there was an effort to not use `github.com/pkg/errors` quite some time ago as Go can wrap / unwrap errors. Would be good to not re-introduce it unless needed :thinking: 
* https://github.com/integr8ly/integreatly-operator/pull/275

KevFan reviewed (commented):
Overall looks good :+1: 

Left some comments on some small things :man_technologist:

laurafitzgerald commented on Dockerfile.osde2e:
```diff
@@ -14,6 +14,8 @@ COPY test ./test
 COPY manifests ./manifests
 COPY version ./version
 COPY Makefile ./
+COPY version ./version
```
I think this got in with the rebase. well spotted. 👍

laurafitzgerald commented:
/test e2e
/test e2e-rhoam

laurafitzgerald commented:
/test rhoam-e2e

matskiv commented on test-cases/tests/installation/a34-verify-sku-values.md:
```diff
@@ -0,0 +1,36 @@
+      - osd-fresh-install
+      - osd-post-upgrade
+    targets:
+      - 1.4.0
```
```suggestion
      - 1.6.0
```
I am not sure if it needs to have "targets" set at all, but if yes, then 1.6.0 probably makes more sense :)

laurafitzgerald replied:
@psturc @trepel would be interested in your thoughts here. I agree we should start from 1.6.0 but should we do it every install. The manual steps are themselves quite small as we've done work to automate the checks needed to a fairly thorough value. The automation is A34.

matskiv commented on test-cases/tests/installation/a34-verify-sku-values.md:
````diff
@@ -0,0 +1,36 @@
+oc get rhmi rhoam -n redhat-rhoam-operator -o json | jq -r '.status.sku'
+```
+
+> The output should be "default" for RHOAM 1.4.0, then FIVE_MILLION_SKU, TWENTY_MILLION_SKU, etc. for next RHOAM releases, based on the SKU type selected in OCM
````
This text probably needs to be updated.

matskiv commented on pkg/resources/quota/quota.go:
```diff
@@ -0,0 +1,348 @@
+	Resources map[string]ResourceConfig     `json:"resources,omitempty"`
+}
+
+func GetQuota(QuotaId string, QuotaConfig *corev1.ConfigMap, retQuota *Quota, isUpdated bool) error {
```
Why are we passing in `retQuota *Quota` instead of returning `Quota` the usual way?

laurafitzgerald replied:
we decided to follow a similar syntax as with the k8s client `Get` functions where a pointer is passed and only an error is returned.

laurafitzgerald commented on pkg/products/rhssouser/reconciler.go:
```diff
@@ -273,14 +273,27 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		},
 	}
+
+	key, err := k8sclient.ObjectKeyFromObject(kc)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	err = serverClient.Get(ctx, key, kc)
+	if err != nil {
+		if !k8serr.IsNotFound(err) {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+	}
```
Hi @matskiv it's an interesting one and well spotted. the CreateOrUpdate does also do a get but inside the createorupdate we are calling the Configure function which requires passing the incluster object. If we don't do the get we only have an in memory object which is not enough to do the configuration required. e.g. we do a check that if the value is greater in the resourcerequirement block we will allow that. We can only do that comparison if we Get the object outside of the create or update.

laurafitzgerald commented on pkg/products/marin3r/rateLimitService.go:
```diff
@@ -146,29 +148,37 @@ func (r *RateLimitServiceReconciler) reconcileDeployment(ctx context.Context, cl
 		},
 	}
 
+	key, err := k8sclient.ObjectKeyFromObject(deployment)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	err = client.Get(ctx, key, deployment)
+	if err != nil {
+		if !k8sError.IsNotFound(err) {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+	}
```
as below.

laurafitzgerald commented on config/configmap/sku-config.yaml:
```diff
@@ -0,0 +1,388 @@
+                    }
+                }
+              }
+          }
```
thanks. at standup we mentioned that @trepel was addressing this as part of another JIRA. @trepel can you confirm?

laurafitzgerald replied:
This will be addressed in -> https://issues.redhat.com/browse/MGDAPI-1632

matskiv commented on pkg/resources/quota/quota.go:
```diff
@@ -0,0 +1,348 @@
+		p.mutateResourcesRequirement(t.Spec.Backend.WorkerSpec.Resources, BackendWorkerName)
+
+	default:
+		return errors.New(fmt.Sprintf("quota configuration can only be applied to Deployments, StatefulSets or Deployment Configs, found %s", reflect.TypeOf(obj)))
```
or Keycloak, or APIManager :)

matskiv commented on pkg/resources/quota/quota_test.go:
```diff
@@ -0,0 +1,960 @@
+	v13 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Double import - `v12 "k8s.io/apimachinery/pkg/apis/meta/v1"`
and also `v13 "k8s.io/api/core/v1"` above

laurafitzgerald commented:
@MStokluska I'm seeing the following error ```[36mINFO[0m[2021-04-26T16:26:33Z] error: "registry.ci.openshift.org/integr8ly/MGDAPI-1192-2:integreatly-operator-test-harness" is not a valid image reference: repository name must be lowercase``` on the failing prow job ```https://storage.googleapis.com/origin-ci-test/logs/branch-ci-integr8ly-integreatly-operator-MGDAPI-1192-3-images/1386715994990317568/build-log.txt``` 
Do you have any concern with this or once it's merged will it be alright?

MStokluska commented:
> @MStokluska I'm seeing the following error `�[36mINFO�[0m[2021-04-26T16:26:33Z] error: "registry.ci.openshift.org/integr8ly/MGDAPI-1192-2:integreatly-operator-test-harness" is not a valid image reference: repository name must be lowercase` on the failing prow job `https://storage.googleapis.com/origin-ci-test/logs/branch-ci-integr8ly-integreatly-operator-MGDAPI-1192-3-images/1386715994990317568/build-log.txt`
> Do you have any concern with this or once it's merged will it be alright?

I don't see that this is being a real issue - we were using capital names before and it never was an issue. @Boomatang do you think this can be related to the go bump(or part of it) that got merged recently?

laurafitzgerald commented:
@KevFan @matskiv i've addressed your feedback here -> https://github.com/integr8ly/integreatly-operator/pull/1740

Shall I go ahead and resolve all the comments in this pr for items addressed or would you prefer to check that and resolve them yourselves?

Boomatang commented:
> > @MStokluska I'm seeing the following error `�[36mINFO�[0m[2021-04-26T16:26:33Z] error: "registry.ci.openshift.org/integr8ly/MGDAPI-1192-2:integreatly-operator-test-harness" is not a valid image reference: repository name must be lowercase` on the failing prow job `https://storage.googleapis.com/origin-ci-test/logs/branch-ci-integr8ly-integreatly-operator-MGDAPI-1192-3-images/1386715994990317568/build-log.txt`
> > Do you have any concern with this or once it's merged will it be alright?
> 
> I don't see that this is being a real issue - we were using capital names before and it never was an issue. @Boomatang do you think this can be related to the go bump(or part of it) that got merged recently?

@MStokluska I would be very surprised if any of the work around the version would affect this PR. The only change that has being made to the operator master branch is to do with the openshift-ci which we got blocked on. There has being no changes made to the openshift/release repo.

KevFan commented:
> @KevFan @matskiv i've addressed your feedback here -> #1740
> 
> Shall I go ahead and resolve all the comments in this pr for items addressed or would you prefer to check that and resolve them yourselves?

@laurafitzgerald Check the PR and looks good :+1: On the resolving comments, could you close resolve them ?  I don't think I have the permissions to resolve them myself :joy: 
* https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#resolving-conversations

matskiv commented on pkg/products/marin3r/rateLimitService.go:
```diff
@@ -146,29 +147,37 @@ func (r *RateLimitServiceReconciler) reconcileDeployment(ctx context.Context, cl
 	}
 
+	key, err := k8sclient.ObjectKeyFromObject(deployment)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
```
If I am looking at the correct source code for `ObjectKeyFromObject()`,
then we don't need any error checking as it never returns errors:
https://github.com/kubernetes-sigs/controller-runtime/blob/b2c90ab82af89fb84120108af14f4ade9df5d787/pkg/client/interfaces.go#L33-L36
And this applies to other `k8sclient.ObjectKeyFromObject()` calls in this PR...

matskiv replied:
Please ignore, I was looking at newer version of controller-runtime.

laurafitzgerald commented:
/test e2e
/test rhoam-e2e

sergioifg94 commented:
/lgtm
/approved

laurafitzgerald commented:
/unhold

sergioifg94 commented:
:rocket: 
/approve
